packages/material-ui/src/Radio/Radio.js

```diff
@@ -49,6 +49,7 @@ function Radio(props) {
         checked: classes.checked,
         disabled: classes.disabled,
       }}
+      role="radio"
```

Comment:
Isn't that duplication?

Reply:
Well it depends what you consider the radio widget. I would consider the e.g. icon that displays the checked status as part of the radio. The input is not what's actually used to display the value of the radio. It's just used to store the value for processing.

Currently we  generate the following markup:
```html
<span>
  <span><svg /><input /></span>
  <span></span>
</span>
```

We even add a visual style to the span if focused and not the input. I'd say it's reasonable to move the radio role up.

Other benefits:
https://deploy-preview-13548--material-ui.netlify.com/demos/selection-controls/#radio-buttons

Focusing the group will enable ChromeVox to determine that the radiogroup has 4 options. Current implementation won't have that which leads me to think that this does help screenreaders. Though it will repeat that the focused element is a radio button

I guess one of the issues is that we currently place the input within the label so fully supporting a11y features is probably out of scope for this PR. I might revisit that in the future and see if we can do better.

At least moving the `switch` role to the input should be a no-brainer.

Reply:
@eps1lon I happy to revisit the approach to help assistive technologies.